src/emoji/data.js

```diff
@@ -1,11 +1,18 @@
 /* @flow strict-local */
-import type { ImageEmojiType, EmojiType, ReactionType } from '../types';
+import * as typeahead from '@zulip/shared/js/typeahead';
+
+import type { EmojiType, ReactionType, LocalEmoji } from '../types';
 import { objectFromEntries } from '../jsBackport';
 import { unicodeCodeByName, override } from './codePointMap';
 import zulipExtraEmojiMap from './zulipExtraEmojiMap';
 import objectEntries from '../utils/objectEntries';
 
 const unicodeEmojiNames = Object.keys(unicodeCodeByName);
+const unicodeEmojiObjects = objectEntries(unicodeCodeByName).map(([name, code]) => ({
+  emoji_name: name,
+  emoji_code: code,
+  emoji_type: 'unicode',
```

Comment:
nit: Because the emoji_code has meaning only within a namespace that's determined by the emoji_type (see jsdoc at our `Reaction` type), the emoji_type should come first.

Same at the type definition for `LocalEmoji` (aka `EmojiForShared` or whatever it might be), and at the other object literals like this one.